src/main/java/co/skyver/employees/service/TitleServiceImpl.java

```diff
@@ -0,0 +1,44 @@
+package co.skyver.employees.service;
+
+import co.skyver.employees.exception.EmployeeNotFoundException;
+import co.skyver.employees.model.Employee;
+import co.skyver.employees.model.Title;
+import co.skyver.employees.repository.EmployeeRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.List;
+
+@Service
+public class TitleServiceImpl implements TitleService {
+
+    private final EmployeeRepository employeeRepository;
+
+    @Autowired
+    public TitleServiceImpl(EmployeeRepository employeeRepository) {
+        this.employeeRepository = employeeRepository;
+    }
+
+    @Override
+    @Transactional
+    public Title assignTitleToEmployee(Long empNo, Title titleToAdd) {
+        Employee employee = this.employeeRepository.findById(empNo).orElseThrow(() -> new EmployeeNotFoundException(empNo));
+        titleToAdd.setEmpNo(empNo);
+        List<Title> titles = employee.getTitles();
+        for (Title title : titles) {
+            if (title.getToDate() == null) {
+                title.setToDate(titleToAdd.getFromDate());
+            }
+        }
```

Comment:
I'd put all this logic inside the `assignTitle` method, so at the end, it would read:

```
Employee employee...
employee.assignTitle(titleToAdd);
this.employeeRepository.save(employee);
```